### PR TITLE
Add more config options

### PR DESCRIPTION
**Pull request by S-S-X**

### CODE REVIEW

File: technic/tools/init.lua [hunk 1]
```diff
@@ -1,24 +1,21 @@
 local path = technic.modpath.."/tools"
 
-if technic.config:get_bool("enable_mining_drill") then
-	dofile(path.."/mining_drill.lua")
+local function enabled(name)
+	return technic.config:get_bool("enable_" .. name)
 end
```
Comment (S-S-X) [inline technic/tools/init.lua:5]: Opinions about this?
Added to make list of loadable files cleaner and bit easier to read.

Comment (OgelGames) [reply technic/tools/init.lua:5]: Makes sense to me.

**Comment by OgelGames** [comment]
Sounds good :+1: One thing that needs to be checked, is if anything depends on those tools (or anything else in the files), such as recipes.

**Comment by S-S-X** [comment]
> Sounds good  One thing that needs to be checked, is if anything depends on those tools (or anything else in the files), such as recipes.
> 
> I think other things could be added too, like how much charge is stored and used by each of the tools.

Only thing I found (internal dependency) was extractor recipe
`technic/machines/LV/extractor.lua:7:            {'technic:treetap', 'basic_materials:motor',          'technic:treetap'},`

Added alternative recipe.

**Comment by S-S-X** [comment]
> I think other things could be added too, like how much charge is stored and used by each of the tools.

Yes that would be good thing but maybe add another PR for that after done with this.

For this PR, while changing configuration a bit I was also thinking about adding few compatibility shims for `technic.config`, for me it is bit weird that minetest config object does not support `get_int`.
Could maybe add that for `technic.config` but for this I think it would be better to crash during loading if config contains invalid number value, and preferably crash with clear error message.

Or could just keep doing that `tonumber(technic.config:get("blablabla"))` thing, problem with that is if it crashes error message most probably is not very clear pointing to configuration error but instead is some less clear nil error or some more obscure weird behavior.

**Comment by S-S-X** [comment]
Busted tests apparently fail in mineunit fake Settings object which is function returning table with getters/setters instead of actual class like object with metatable.
Problem is in mineunit and should be fixed there: make mineunit Settings to actually behave and look like Minetest Settings object.

File: technic/config.lua [hunk 2]
```diff
@@ -1,28 +1,74 @@
...
+local config_obj_mt = getmetatable(config_obj)
+for key,value in pairs(config_obj_mt) do
+	if type(value) == "function" then
+		Config[key] = function(self, ...) return value(config_obj, ...) end
+	else
+		config_obj_mt[key] = value
+	end
+end
+setmetatable(Config, config_obj_mt)
+technic.config = setmetatable({}, Config)
```
Comment (S-S-X) [inline technic/config.lua:69]: Not sure if this is a bit over-engineered. Would it be better to just define `technic.config` object to be table with functions that either forward calls to underlying Settings object (`get`, `write`, `to_table`, etc.) or implement call (`get_int`)?
I feel this is over-engineered as in overly complex solution to very simple problem...

Comment (BuckarooBanzay) [reply technic/config.lua:69]: uhm, i haven't checked every setting related code, but why don't we use hardcoded setting-keys (`technic.*`) and `minetest.settings:get_*` everywhere?

Comment (S-S-X) [reply technic/config.lua:69]: Well, `minetest.settings` is `Settings` object and therefore `:get_*` does not include `get_int` which was one attempted improvement here.
Everything or at least almost everything is in `worldpath/technic.conf` already.

`technic.config` is `Settings` object and therefore has exact same API that `minetest.settings` offers.

Some maybe about 20 new settings would be added on top of current about 20 options, none of those are currently prefixed in any way and I think count is high enough and options specific enough to warrant using separate configuration file.

Comment (S-S-X) [reply technic/config.lua]: Updated technic.config forwarders to be significantly simpler and easier to understand.

**Comment by S-S-X** [comment]
lol :smile: that already catched my failure with config key

**Comment by BuckarooBanzay** [comment]
should've squashed that, sorry :facepalm:

**Pull request merged**